pkg/framework/context/context.go

```diff
@@ -231,18 +231,13 @@ func WithPackager(p commontransport.Packager) ProviderOption {
 	}
 }
 
-// WithPacker injects a Packer into the context as the outbound Packer
-func WithPacker(p packer.Packer) ProviderOption {
+// WithPacker injects at least one Packer into the context,
+// with the primary Packer being used for inbound/outbound communication
+// and the additional packers being available for unpacking inbound messages.
+func WithPacker(primary packer.Packer, additionalPackers ...packer.Packer) ProviderOption {
 	return func(opts *Provider) error {
-		opts.packer = p
-		return nil
-	}
-}
-
-// WithInboundPackers injects a variable number of Packer services into the Aries framework
-func WithInboundPackers(packers ...packer.Packer) ProviderOption {
-	return func(opts *Provider) error {
-		opts.inboundPackers = append(opts.inboundPackers, packers...)
+		opts.packer = primary
```

Comment:
thinking about this.. primaryPacker is not optional.. it is mandatory
or we can say it's mandatory if WithPackers() is not used

another thought:
If WithPacker is not set and WithPackers is set, then use the first packer in the WithPackers list as primary packer. There must always be one packer created regardless if both With options were used or not (hence fallback to default packer).

Reply:
There should be no more `WithPackers()` - `WithPacker()` takes extra packers as variadic args after the primary one.